common/src/autogluon/common/loaders/load_s3.py

```diff
@@ -27,47 +29,85 @@ def download(input_bucket, input_prefix, local_path):
     s3.Bucket(input_bucket).download_file(input_prefix, local_path)
 
 
-def list_bucket_prefix_s3(bucket, prefix):
-    return list_bucket_prefix_suffix_s3(bucket=bucket, prefix=prefix)
-
-
-def list_bucket_prefix_suffix_s3(bucket, prefix, suffix=None, banned_suffixes=None):
+# TODO: Expand to support arbitrary list of candidate file paths
+# TODO: add local file path support
+# TODO: Consider renaming and deprecating old name
+def list_bucket_prefix_suffix_contains_s3(bucket: str,
+                                          prefix: str,
+                                          suffix: Union[str, List[str]] = None,
+                                          banned_suffixes: List[str] = None,
+                                          contains: Union[str, List[str]] = None,
+                                          banned_contains: List[str] = None) -> List[str]:
```

Comment:
since you are using `None` as default, this should be:
```python
suffix: Optional[Union[str, List[str]]] = None,
banned_suffixes: Optional[List[str]] = None,
contains: Optional[Union[str, List[str]]] = None,
banned_contains: Optional[List[str]] = None,
```

Reply:
updated